<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-05-02">

<title>Find a Gene Project Q1-Q4</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="Find a Gene Q1-Q4_files/libs/clipboard/clipboard.min.js"></script>
<script src="Find a Gene Q1-Q4_files/libs/quarto-html/quarto.js"></script>
<script src="Find a Gene Q1-Q4_files/libs/quarto-html/popper.min.js"></script>
<script src="Find a Gene Q1-Q4_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Find a Gene Q1-Q4_files/libs/quarto-html/anchor.min.js"></script>
<link href="Find a Gene Q1-Q4_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Find a Gene Q1-Q4_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Find a Gene Q1-Q4_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Find a Gene Q1-Q4_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Find a Gene Q1-Q4_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Find a Gene Project Q1-Q4</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 2, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="find-a-gene" class="level1">
<h1>Find a Gene</h1>
<section id="q1-tell-me-the-name-of-a-protein-you-are-interested-in.-include-the-species-and-the-accession-number.-this-can-be-a-human-protein-or-a-protein-from-any-other-species-as-long-as-its-function-is-known." class="level3">
<h3 class="anchored" data-anchor-id="q1-tell-me-the-name-of-a-protein-you-are-interested-in.-include-the-species-and-the-accession-number.-this-can-be-a-human-protein-or-a-protein-from-any-other-species-as-long-as-its-function-is-known.">[<strong>Q1</strong>] Tell me the name of a protein you are interested in. Include the species and the accession number. This can be a human protein or a protein from any other species as long as itâ€™s function is known.</h3>
<p>Name: Synaptotagmin-1, isoform 1</p>
<p>Accession: NP_001285568</p>
<p>Species: Drosophila melanogaster (flies)</p>
</section>
<section id="q2-perform-a-blast-search-against-a-dna-database-such-as-a-database-consisting-of-genomic-dna-or-ests.-the-blast-server-can-be-at-ncbi-or-elsewhere.-include-details-of-the-blast-method-used-database-searched-and-any-limits-applied-e.g.-organism." class="level3">
<h3 class="anchored" data-anchor-id="q2-perform-a-blast-search-against-a-dna-database-such-as-a-database-consisting-of-genomic-dna-or-ests.-the-blast-server-can-be-at-ncbi-or-elsewhere.-include-details-of-the-blast-method-used-database-searched-and-any-limits-applied-e.g.-organism.">[<strong>Q2</strong>] Perform a BLAST search against a DNA database, such as a database consisting of genomic DNA or ESTs. The BLAST server can be at NCBI or elsewhere. Include details of the BLAST method used, database searched and any limits applied (e.g.&nbsp;Organism).</h3>
<p>Method: blastp search against mus musculus (mice) proteins</p>
<p>Database: RefSeq select proteins (refseq_select)</p>
<p>Organism: Mus Musculus (Taxid: 10090)</p>
<p><img src="images/8EAEFDAE-30B3-4063-B0D6-EA1E010CE1D2.jpeg" class="img-fluid"></p>
<p>Chosen match: Accession NP_033332.1, a 421 base pair clone from Mus musculus. See below for alignment details.</p>
<p><img src="images/6290C16F-BA1B-4850-8715-ADBCE5FCC297.jpeg" class="img-fluid"></p>
<p><img src="images/7B41D049-F90E-4DA7-AD5C-8AF87EE7A5EE.jpeg" class="img-fluid"></p>
<p>Alignment details:</p>
<p>synaptotagmin-1 isoform 1 [Mus musculus]</p>
<p>Sequence ID:&nbsp;<a href="https://www.ncbi.nlm.nih.gov/protein/NP_033332.1?report=genbank&amp;log$=protalign&amp;blast_rank=1&amp;RID=52390PZM016" title="Show report for NP_033332.1">NP_033332.1</a> Length:&nbsp;421 Number of Matches:&nbsp;1</p>
<table class="table">
<colgroup>
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 20%">
<col style="width: 16%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Score</th>
<th style="text-align: left;">Expect</th>
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Identities</th>
<th style="text-align: left;">Positives</th>
<th style="text-align: left;">Gaps</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">424 bits(1090)</td>
<td style="text-align: left;">3e-146</td>
<td style="text-align: left;">Compositional matrix adjust.</td>
<td style="text-align: left;">247/418(59%)</td>
<td style="text-align: left;">303/418(72%)</td>
<td style="text-align: left;">12/418(2%)</td>
</tr>
</tbody>
</table>
<pre><code>Query  60   AASQRIAQVESTTRSATTEAQESTTTAVPVIKKIEHVGEVVTEVIAE--RTGLPTWGVVA  117
            +AS+  A     T  AT     +T  A P   K +   ++  + + E  +  LP W ++A
Sbjct  3    SASRPEALAAPVTTVATLVPHNATEPASPGEGKEDAFSKLKQKFMNELHKIPLPPWALIA  62

Query  118  IIILVFLVVFGIIFFCV-----RRFLKKRRTKDGKGKKGVDMKSVQLLGSAYKEKP--DM  170
            I I+  L+V    F CV      +   K++ K+  GK  ++MK V+ LG   K++   D 
Sbjct  63   IAIVAVLLVVTCCF-CVCKKCLFKKKNKKKGKEKGGKNAINMKDVKDLGKTMKDQALKDD  121

Query  171  EELTENAEEGDEEDKQSEQKLGRLNFKLEYDFNSNSLAVTVIQAEELPALDMGGTSDPYV  230
            +  T   +  ++E+ + E+KLG+L + L+YDF +N L V +IQA ELPALDMGGTSDPYV
Sbjct  122  DAETGLTDGEEKEEPKEEEKLGKLQYSLDYDFQNNQLLVGIIQAAELPALDMGGTSDPYV  181

Query  231  KVYLLPDKKKKFETKVHRKTLSPVFNETFTFKSLPYADAMNKTLVFAIFDFDRFSKHDQI  290
            KV+LLPDKKKKFETKVHRKTL+PVFNE FTFK +PY++   KTLV A++DFDRFSKHD I
Sbjct  182  KVFLLPDKKKKFETKVHRKTLNPVFNEQFTFK-VPYSELGGKTLVMAVYDFDRFSKHDII  240

Query  291  GEVKVPLCTIDLAQTIEEWRDLVSVEGEGGQEKLGDICFSLRYVPTAGKLTVVILEAKNL  350
            GE KVP+ T+D     EEWRDL S E E  QEKLGDICFSLRYVPTAGKLTVVILEAKNL
Sbjct  241  GEFKVPMNTVDFGHVTEEWRDLQSAEKEE-QEKLGDICFSLRYVPTAGKLTVVILEAKNL  299

Query  351  KKMDVGGLSDPYVKIAIMQNGKRLKKKKTSIKKCTLNPYYNESFSFEVPFEQIQKICLVV  410
            KKMDVGGLSDPYVKI +MQNGKRLKKKKT+IKK TLNPYYNESFSFEVPFEQIQK+ +VV
Sbjct  300  KKMDVGGLSDPYVKIHLMQNGKRLKKKKTTIKKNTLNPYYNESFSFEVPFEQIQKVQVVV  359

Query  411  TVVDYDRIGTSEPIGRCILGCMGTGTELRHWSDMLASPRRPIAQWHTLKDPEETDEIL  468
            TV+DYD+IG ++ IG+  +G   TG ELRHWSDMLA+PRRPIAQWHTL+  EE D +L
Sbjct  360  TVLDYDKIGKNDAIGKVFVGYNSTGAELRHWSDMLANPRRPIAQWHTLQVEEEVDAML  417</code></pre>
<p>[<strong>Q3</strong>] Gather information about this "novel" <strong>protein</strong>. At a minimum, show me the protein sequence of the "novel" protein as displayed in your BLAST results from [Q2] as FASTA format (you can copy and paste the aligned sequence subject lines from your BLAST result page if necessary) or translate your novel DNA sequence using a tool called EMBOSS Transeq at the EBI. Don't forget to translate all six reading frames; the ORF (open reading frame) is likely to be the longest sequence without a stop codon. It may not start with a methionine if you don't have the complete coding region. Make sure the sequence you provide includes a header/subject line and is in traditional FASTA format.</p>
<p>Chosen sequence:</p>
<pre><code>&gt;NP_033332.1 synaptotagmin-1 isoform 1 [Mus musculus]
MVSASRPEALAAPVTTVATLVPHNATEPASPGEGKEDAFSKLKQKFMNELHKIPLPPWALIAIAIVAVLL
VVTCCFCVCKKCLFKKKNKKKGKEKGGKNAINMKDVKDLGKTMKDQALKDDDAETGLTDGEEKEEPKEEE
KLGKLQYSLDYDFQNNQLLVGIIQAAELPALDMGGTSDPYVKVFLLPDKKKKFETKVHRKTLNPVFNEQF
TFKVPYSELGGKTLVMAVYDFDRFSKHDIIGEFKVPMNTVDFGHVTEEWRDLQSAEKEEQEKLGDICFSL
RYVPTAGKLTVVILEAKNLKKMDVGGLSDPYVKIHLMQNGKRLKKKKTTIKKNTLNPYYNESFSFEVPFE
QIQKVQVVVTVLDYDKIGKNDAIGKVFVGYNSTGAELRHWSDMLANPRRPIAQWHTLQVEEEVDAMLAVK
K</code></pre>
<p><strong>Name</strong>: synaptotagmin-1 isoform 1 [Mus musculus] <strong>Species</strong>:</p>
<pre><code>Mus musculus
            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;
            Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha;
            Muroidea; Muridae; Murinae; Mus; Mus.</code></pre>
<p>[<strong>Q4</strong>] Prove that this gene, and its corresponding protein, are novel. For the purposes of this project, "novel" is defined as follows. Take the protein sequence (your answer to [Q3]), and use it as a query in a blastp search of the nr database at NCBI.</p>
<p><img src="images/15A5A3F9-BB23-483F-AE5E-FCECF258CA9C.jpeg" class="img-fluid"></p>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>